# %YAML:1.0
#########################################################################
# ⭐⭐⭐  Set parameters carefully for satisfied performance!   ⭐⭐⭐ #
#########################################################################

mass        : 1.4 # kg 
gra         : 9.805 
pose_solver : 1     # 0:From ZhepeiWang (drag & less singular) 1:From ZhepeiWang, 2:From rotor-drag    
ctrl_freq_max   : 50.0
use_bodyrate_ctrl: false
max_manual_vel: 1
max_angle: 30  # Attitude angle limit in degree. A negative value means no limit.
low_voltage: 13.2 # 4S battery
body_T_cam: "/home/luochen/drone-project/drone-project/src/realflight_modules/px4ctrl/config/body_T_cam0_xtdrone.yaml"
body_T_camdown: "/home/luochen/project_doc/drone_msproject/drone/src/realflight_modules/px4ctrl/config/body_T_camdown_xtdrone.yaml"
# body_T_cam: "body_T_cam0_xtdrone.yaml"
rc_reverse: # *
    roll: false
    pitch: false
    yaw: false
    throttle: false

auto_takeoff_land:
    land_on_target: true
    enable: true
    enable_auto_arm: true
    no_RC: true
    takeoff_height: 5.5 # m
    takeoff_land_speed: 0.4 # m/s
    fly_speed: 0.6 # m/s

thrust_model: # The model that maps thrust signal u(0~1) to real thrust force F(Unit:N): F=K1*Voltage^K2*(K3*u^2+(1-K3)*u). 
    print_value: true # display the value of “thr_scale_compensate” or “hover_percentage” during thrust model estimating.
    accurate_thrust_model: true  # This can always enabled if don't require accurate control performance :-)
    # accurate thrust mapping parameters
    K1: 0.7583 # Needs precise calibration!
    K2: 1.6942 # Needs precise calibration!
    K3: 0.6786 # Needs precise calibration! K3 equals THR_MDL_FAC in https://docs.px4.io/master/en/config_mc/pid_tuning_guide_multicopter.html.
    # approximate thrust mapping parameters
    hover_percentage: 0.385  # Thrust percentage in Stabilize/Arco mode # *

gain: 
    # Cascade PID controller. Recommend to read the code.
    Kp0: 2  #2
    Kp1: 2  #2
    Kp2: 2  #2
    Kv0: 3  #3
    Kv1: 3  #3
    Kv2: 3  #3
    # ↓↓↓ No use now --
    Kvi0: 0.05  #0.05
    Kvi1: 0.05  #0.05
    Kvi2: 0.05  #0.05
    Kvd0: 0.3
    Kvd1: 0.3
    Kvd2: 0.3
    # ↓↓↓ Only used in rate control mode.
    KAngR: 20.0
    KAngP: 20.0
    KAngY: 20.0

rotor_drag:  
    x: 0.0  # The reduced acceleration on each axis caused by rotor drag. Unit:(m*s^-2)/(m*s^-1).
    y: 0.0  # Same as above
    z: 0.0  # Same as above
    k_thrust_horz: 0.0 # Set to 0 recommended... --

msg_timeout:
    odom: 0.5
    rc:   0.5
    cmd:  0.5
    imu:  0.5
    bat:  0.5

disturbance_obs:
    constant: false # net output set to constant
    use: true     # use in controller
    R: 40          
    Q: 2.0
    P: 1.0
    lamda: 0.02    #0.02效果很好
    


csv_filename:   "/home/luochen/project_doc/drone_msproject/neural-fly/sim_data_-x/experiment/data.csv"
prefix: "/iris_0"
model_path: "/home/luochen/project_doc/drone_msproject/neural-fly/net.pt"

